source/tests/system/nirfmxgsm_session_tests.cpp

```diff
@@ -0,0 +1,185 @@
+  }
+}
+
+TEST_F(NiRFmxGSMSessionTest, InitializedSession_CloseSession_ClosesDriverSession)
+{
+  rfmxgsm::InitializeResponse init_response;
+  call_initialize(kRFmxGSMTestRsrc, kRFmxGSMOptionsString, kRFmxGSMTestSessionOne, &init_response);
+
+  nidevice_grpc::Session session = init_response.instrument();
+  ::grpc::ClientContext context;
+  rfmxgsm::CloseRequest close_request;
+  close_request.mutable_instrument()->set_name(session.name());
+  close_request.set_force_destroy(false);
+  rfmxgsm::CloseResponse close_response;
+  ::grpc::Status status = GetStub()->Close(&context, close_request, &close_response);
+
+  EXPECT_TRUE(status.ok());
+  EXPECT_EQ(0, close_response.status());
+}
+
+TEST_F(NiRFmxGSMSessionTest, TwoInitializedSessionsOnSameDevice_CloseSessions_ClosesDriverSessions)
+{
+  rfmxgsm::InitializeResponse init_response_one, init_response_two;
+  ::grpc::Status status_one = call_initialize(kRFmxGSMTestRsrc, kRFmxGSMOptionsString, kRFmxGSMTestSessionOne, &init_response_one);
+  ::grpc::Status status_two = call_initialize(kRFmxGSMTestRsrc, kRFmxGSMOptionsString, kRFmxGSMTestSessionTwo, &init_response_two);
+  EXPECT_TRUE(status_one.ok());
+  EXPECT_EQ(0, init_response_one.status());
+  EXPECT_TRUE(status_two.ok());
+  EXPECT_EQ(0, init_response_two.status());
+  EXPECT_NE(init_response_one.instrument().name(), init_response_two.instrument().name());
+
+  nidevice_grpc::Session session_one = init_response_one.instrument();
+  nidevice_grpc::Session session_two = init_response_two.instrument();
+  rfmxgsm::CloseResponse close_response_one, close_response_two;
+  status_one = call_close(session_one, false, &close_response_one);
+  status_two = call_close(session_two, false, &close_response_two);
+
+  EXPECT_TRUE(status_one.ok());
+  EXPECT_EQ(0, close_response_one.status());
+  EXPECT_TRUE(status_two.ok());
+  EXPECT_EQ(0, close_response_two.status());
+}
+
+TEST_F(NiRFmxGSMSessionTest, CallInitializeTwiceWithSameSessionNameOnSameDevice_CloseSessionTwice_SecondCloseReturnsInvalidSessionError)
+{
+  rfmxgsm::InitializeResponse init_response_one, init_response_two;
+  ::grpc::Status status_one = call_initialize(kRFmxGSMTestRsrc, kRFmxGSMOptionsString, kRFmxGSMTestSessionOne, &init_response_one);
+  ::grpc::Status status_two = call_initialize(kRFmxGSMTestRsrc, kRFmxGSMOptionsString, kRFmxGSMTestSessionOne, &init_response_two);
+  EXPECT_TRUE(status_one.ok());
+  EXPECT_EQ(0, init_response_one.status());
+  EXPECT_TRUE(status_two.ok());
+  EXPECT_EQ(0, init_response_two.status());
+  EXPECT_EQ(init_response_one.instrument().name(), init_response_two.instrument().name());
+
+  nidevice_grpc::Session session_one = init_response_one.instrument();
+  nidevice_grpc::Session session_two = init_response_two.instrument();
+  rfmxgsm::CloseResponse close_response_one, close_response_two;
+  status_one = call_close(session_one, false, &close_response_one);
+  EXPECT_TRUE(status_one.ok());
+  EXPECT_EQ(0, close_response_one.status());
+
+  try {
```

Comment:
More of an FYI as I think this is fine, but you can use EXPECT_THROW as an alternative to a try-catch with a `FAIL()` assertion. It gets kind of messy if you want to validate properties on the exception but here's [an example](https://github.com/ni/grpc-device/blob/ca6a68aa7a18f41419e33f70c75ac9c95e9ad732/source/tests/unit/session_repository_tests.cpp#L162) in some of our unit tests that do it.

Reply:
I'm not sure I could use EXPECT_THROW in any of these places without it getting a little messy.  The current session tests for other RFmx personalities also use FAIL() in this way, so I'm going to leave this as is.